### ISSUE TITLE
"This action could not be completed due to possible environment mismatch"

### ISSUE THREAD

**Issue by Nosskirneh** [NyaMisty/AltServer-Linux#105]
After upgrading to AltStore 1.7, refreshing an app with AltServer-Linux yields in the error with message "This action could not be completed due to possible environment mismatch" and error code -29004.

**Comment by Rattlyy** [comment]
same issue

**Comment by CrCrate** [comment]
same issue, using  2.0b2 (iOS 16.1.2, iPhone13,1) but it worked fine before so i dont it’s the altstore version at fault. likely related to #104 as they both seemed to have popped up at the same time and the only solutions i could find to -29004 on altserver(windows) also involved the adi folder. 

(i was also unable to refresh apps using any other altserver, untouched previously working linux or windows all cause the same issue somehow, perhaps entirely unrelated to altserverlinux?)

**Comment by babaruh** [comment]
yes, the first reason why I tried to reinstall altserver is that I can't refresh applications (so it's related to #104) . keep in mind that it worked before, so something has changed but i don't know what

**Comment by Whitetigerswt** [comment]
Has anyone found a solution to this yet?

**Comment by babaruh** [comment]
@Whitetigerswt I haven't found a solution that solves the problem completely, but using an alternative Anisette server I was able to install ipa (although the altstore is not working). you can find solution in #104

**Comment by codingnooob** [comment]
Getting this issue on Linux Mint

**Comment by yusefgharib** [comment]
### Possible Solution:
If anyone is having this issue, I am using a MacBook and I was able to solve it by signing in to iCloud on my mac first. I also turned off find my on the phone and mac, but I'm not sure if that is what fixed it. You need to be signed in to your iCloud on both the iPhone and the MacBook if you're using a MacBook. If you're using Windows, I'm not sure but keep the above in mind and you might be able to solve it!
